src/deploy/AWSDeployer.js

```diff
@@ -546,6 +577,63 @@ export default class AWSDeployer extends BaseDeployer {
     this.log.info(chalk`{green ok}: parameters updated.`);
   }
 
+  async cleanup() {
+    const { cfg, functionName } = this;
+    const cleanupMatchers = [{
+      name: 'CI',
+      property: 'cleanupCiAge',
+      match: (alias) => alias.match(/^ci(\d+)$/),
+    }, {
+      name: 'patch',
+      property: 'cleanupPatchAge',
+      match: (alias) => alias.match(/^(\d+)_(\d+)_(\d+)(-test)?$/),
+    }];
+
+    try {
+      this.log.info('Clean up old aliases and versions');
+      this.log.info(chalk`--: fetching aliases...`);
+      const aliases = await this.listAliases(functionName);
+      this.log.info(chalk`--: fetching versions...`);
+      const versions = (await this.listVersions(functionName))
+        .map((version) => ({
+          ...version,
+          Aliases: aliases
+            .filter(({ FunctionVersion }) => version.Version === FunctionVersion)
+            .map(({ Name }) => Name),
+        }));
+
+      for (const { name, property, match } of cleanupMatchers) {
+        if (cfg[property]) {
+          const cleanupBefore = Date.now() - cfg[property];
+          const oldVersions = versions
+            .filter(({ Aliases }) => Aliases.length > 0 && Aliases.every((alias) => match(alias)))
```

Comment:
we remove a version only, when *every* alias it has does match, so 2 aliases `ci123` and `ci` pointing to the same version will not be removed